### ISSUE TITLE
[IMPROVEMENT] Display 'Out of Stock' on Catalog pages

### ISSUE THREAD

**Issue by Bewitching-Co** [PrestaShop/PrestaShop#21105]
#### Is your feature request related to a problem?

When I am looking at a list of products in a category, there is no indication of the products status unless you click quick view or go directly to the products page. While scrolling thru a list of products in a category, there is NO indication of whether a product is in stock or out of stock without an additional action, either via quick view, or clicking on the product and going to that product's page. Labels are enabled in Back office.

#### Describe the solution you'd like
A simple label on each product while in list view indicating that a product is in stock or out of stock, like ver 1.6 had.
I thought this was included on stock ver 1.7, but apparently it is not.

#### Specs: 
Add a toggle on the **Shop Parameters > Products settings > Products stock** page:
"Display out-of-stock label on product listing pages" in _Admin.Shopparameters.Feature_: Yes/no
Help text under the toggle: "Note that the label will be displayed only if backorders are denied." in _Admin.Shopparameters.Help_
The toggle is on yes by default
The out of stock label should be displayed only if the product stock is <= 0 AND if the order of out of stock product is denied.
If the product has combinations, the out of stock label should be displayed only if all combinations are out of stock (stock is <= 0 AND if the order of out of stock product is denied)

**Be carefull there is 2 ways to define if the order of out of stock product is allowed or not:**
- In Shop Parameters > Products settings > Allow ordering of out-of-stock products
- In the product page > Quantities or combinations tab > Availability preferences

https://github.com/PrestaShop/prestashop-specs/pull/153/commits/310d4a9f03b0c752f15064fbbc649095fb88e978

#### Additional context

![Red Stock indicator](https://user-images.githubusercontent.com/71606342/93886166-b33d4b80-fcaa-11ea-8cc8-63659f517d2c.jpg)
Version 1.6

![No Stock indicator](https://user-images.githubusercontent.com/71606342/93886192-bb958680-fcaa-11ea-9ec2-ed3b8b1ac65d.jpg)
Version 1.7

**Comment by sarahdib** [comment]
@PrestaShop/product-team Could you check this improvement ?
Thank you!

**Comment by prestascott** [comment]
Hi @Bewitching-Co.

Thank you for your improvement request!
This is a nice idea, I'll add it to the related Epic. :) 

Not showing that the product is 'Out of Stock' on this page is a really bad UX. It's as if a merchant is putting an item in its shelves but doesn't sell it. Customers of the merchants will have a negative experience towards their online shopping on merchants' e-ecommerce website. The availability of the product must be clear on this page and on the product page. 

Here is my design proposition where we can use labels on 1.7:
![outofstock-1](https://user-images.githubusercontent.com/54888056/97976947-b4928580-1dcb-11eb-9e3d-142594d0edbb.png)

We can even emphasize the fact that the product is 'Out of Stock' by reducing the image opacity:
![outofstock-2](https://user-images.githubusercontent.com/54888056/97977033-d855cb80-1dcb-11eb-877b-494ca4cd5db2.png)

**Comment by MatShir** [comment]
Indeed it is a nice idea, we just need to keep in mind to make it optional.

**Comment by NeOMakinG** [comment]
@PrestaShop/prestashop-maintainers I need a backend dev for the "optional" feature, I can do the front.

@PrestaShop/product-team How do we display the out of stock label?

**Comment by marionf** [comment]
It's rather a question for @PrestaShop/ux-team

**Comment by prestascott** [comment]
Hi everyone,

Not showing that the product is 'Out of Stock' on this page is a really bad UX. It's as if a merchant is putting an item in its shelves but doesn't sell it. Customers of the merchants will have a negative experience towards their online shopping on merchants' e-ecommerce website. The availability of the product must be clear on this page and on the product page. 

But like @MatShir said, we still need to make it optional for our users.

Here is my design proposition where we can use labels on 1.7:
![outofstock-1](https://user-images.githubusercontent.com/54888056/97976947-b4928580-1dcb-11eb-9e3d-142594d0edbb.png)

We can even emphasize the fact that the product is 'Out of Stock' by reducing the image opacity:
![outofstock-2](https://user-images.githubusercontent.com/54888056/97977033-d855cb80-1dcb-11eb-877b-494ca4cd5db2.png)

**Comment by Progi1984** [comment]
@NeOMakinG Backend in da place ;)

**Comment by NeOMakinG** [comment]
@Progi1984 we should add a toggle on `/configure/shop/product-preferences/` on `Product stock` part, `Show out of stock label on listing pages`, and I should have access to this parameter variable inside smarty

**Comment by Progi1984** [comment]
@NeOMakinG I create the PR this afternoon :) and you continue on it :+1:

**Comment by marionf** [comment]
Maybe we can display another wording than "out of stock" if the product has combinations and is available in other combinations like "available with different options" ? Otherwise I am afraid the customer thinks the product isn't available at all while it's only the default combination

**Comment by Progi1984** [comment]
@marionf What is the default value for "Show out of stock label on listing pages" ?

**Comment by marionf** [comment]
It should be enabled by default, I completed the specs part of the issue.

@Julievrz wdyt of the new label "Display the "out of stock" label on the product's listing pages" and the help text under the toggle "Displayed only if the order of out of stock product is denied." ?

**Comment by Julievrz** [comment]
I suggest the following wording: 

"Display out-of-stock label on product listing pages" in Admin.Shopparameters.Feature
"Note that the label will be displayed only if backorders are denied." in Admin.Shopparameters.Help

**Comment by Bewitching-Co** [comment]
Out of stock already shows on individual product listing page.
It needs to be shown anywhere a group of products are shown ie: category page, featured product on home page etc..
If a product has options, then perhaps "view available options"?


> I suggest the following wording:
> 
> "Display out-of-stock label on product listing page" in Admin.Shopparameters.Feature
> "Note that the label will be displayed only if backorders are denied." in Admin.Shopparameters.Help

**Comment by Julievrz** [comment]
> Out of stock already shows on individual product listing page.
> It needs to be shown anywhere a group of products are shown ie: category page, featured product on home page etc..
> If a product has options, then perhaps "view available options"?
> 
> > I suggest the following wording:
> > "Display out-of-stock label on product listing page" in Admin.Shopparameters.Feature
> > "Note that the label will be displayed only if backorders are denied." in Admin.Shopparameters.Help

I think the "product listing page" is the page where all the products are listed and the product page is the individual page.

"products listing page" is not grammatically correct in my opinion.

Or maybe we could add an "s" to "page" if it applies to multiple pages (category page, featured product, ...) 

If the product has options, I think @marionf 's wording was good "Available with different options".

**Comment by marionf** [comment]
Yes, it's rather "products listing pages" as we can have listing of products on differents pages: categories pages, home page, best sales pages, new products page, prices drop page ...

For products with combinations we will display "out of stock" label only if all combinations are out of stock, otherwise we will display nothing as the wording "Available with different options" is too long to be displayed on the product's image.
Furthermore, the goal is to encourage visitors to click on the product to see all the combinations and choose the one he/she wants. 
Anyway, he/she will be able to see if the selected combination is available or not on the product page.

**Comment by Julievrz** [comment]
@marionf I modified my first comment and added an "s" to "page". However, I'm really not sure about adding an "s" to "product" as I think it is not grammatically correct to write "products listing pages".

**Comment by maciekwyka** [comment]
Should be also option to completely hide products out of stock from shop

**Comment by Hlavtox** [comment]
@marionf @NeOMakinG 

This is easy to implement and extremely important, I think it should definitely be in the template as standard. :-) Me personally, when shopping online, the stock availability is the number 1 option by which I choose where to order from.

And regarding the wording, this field should display **product.available_later** text, which you set in product settings. Every product has different availability options, when it's out of stock.

**IMHO, the logic present in Prestashop 1.6 was working perfectly:**
quantity 1 or more = show _X pieces in stock_
quantity less or 0 = show **product.available_later**, if not set, show "Out of stock"

![availability](https://user-images.githubusercontent.com/6097524/109831339-9e90b680-7c3f-11eb-91cb-91b9434bd3ba.JPG)

**Comment by Hlavtox** [comment]
@NeOMakinG You can just add this under product price like I do on my stores. It works, looks nice and requires no core modification. 👍  See https://eshop-franke.cz/459-stojankove-se-sprchou?order=product.name.asc
								
```
<div class="product-delivery">
    {if $product.quantity > 0}
        <span class="product-delivery-stock">
            {$product.quantity} in stock
        </span>			
    {else}
        <span class="product-delivery-outofstock">
            {if $product.available_later && $product.available_later != ""}
                {$product.available_later}
            {else}
                 Out of stock
             {/if}
        </span>
    {/if}
</div>	
```

**Comment by NeOMakinG** [comment]
Thanks for your feedback

@prestascott since we already added labels, wdyt of @Hlavtox messages ?

**Comment by Hlavtox** [comment]
@prestascott @NeOMakinG Flags (labels) could be a good way to display this, but it only for the bad information, and with limited space. 

I think it is very good to also display that everything is good and how many pieces are in stock. So that's why I would go for a permanent space under product price to display this.

**Comment by Hlavtox** [comment]
The biggest czech ecommerce store with 45k orders per day:
![Výstřižek](https://user-images.githubusercontent.com/6097524/109837793-d3a00780-7c45-11eb-9c34-e6f53813eac6.JPG)

**Comment by NeOMakinG** [comment]
I agree, but I'm not alone to decide on this, PM and UX are needed :)

**Comment by MatShir** [comment]
Indeed, it is a nice suggestion @Hlavtox, but the PR link to the issue had already back and forth with the review. Better to finish what have been started. Your feature request has to be review as well, so IMO let's go with this feature as it is now. @Hlavtox you could create a new issue with your request :)

**Comment by prestascott** [comment]
Good idea @Hlavtox but we should keep it optional. I trust you that the biggest czech e-commerce generates 45k orders a day, no doubt. But not all e-commerce wants to show quantity availability.

**Comment by Progi1984** [comment]
PR Merged with #21751
